Implement Admin role to the backend to protect routes

chrispinkney opened the pull request:
Closes #19, #25

# Description
This PR extends the `User` table by adding a `Role` column, generated from the `Role` enum which can be defined as either `USER` or `ADMIN`.

Users by default are created as `USER` and so far only a SQL `UPDATE` command can create `ADMIN` users and will probably stay that way.

The purpose for this is to defend our `DELETE` routes such-that normal users cannot delete other users- only admins can delete. I added a guard called `AuthGuard` to restrict such action which checks the User's role before letting them hit the endpoint. The User's role is assigned by a new `CurrentUserMiddleware` middleware. The middleware runs on every route- so while expensive (db query)- should solve a lot of problems.

**Update**: 
I've updated this PR to allow `User`s to delete their own account:
- A `User` should only be able to delete their own account now, and nobody else's. An Admin can still delete anyone- including themselves.
- I've also changed the delete route to sign the user out who deleted themselves as I ran into a bug testing this and had to forcibly delete my own user cookie.

# Testing
- `npm run db:generate`

To test, before pulling the PR you can try to delete a user (_might have to create them beforehand since the DB was reset due to changing the schema_) as another user which should work just fine.

Pull the PR and try to delete again which should be blocked. Then try to delete another user as an admin which should work (I'll generate two admin accounts).

Thinbishop reviewed (changes requested):
Tested multiple user deletions with `DEL http://jonvm:3000/auth/:id`. Unable to determine whether user can still delete their own account. Suggesting to add a "delete by email" request (#25) before merging so the admin role changes can be properly tested.

chrispinkney commented:
> Tested multiple user deletions with `DEL http://jonvm:3000/auth/:id`. Unable to determine whether user can still delete their own account. Suggesting to add a "delete by email" request (#25) before merging so the admin role changes can be properly tested.

Great idea! I've updated this PR to include the functionality (inside of the `AdminGuard` which simply returns true if the user is trying to delete their own id).

A `User` should only be able to delete their own account now, and nobody else's. An `Admin` can still delete anyone- including themselves.

I've also changed the delete route to sign the user out who deleted themselves as I ran into a bug testing this and had to forcibly delete my own user cookie.

Thinbishop reviewed (approved):
Tested signing in, trying to delete other user (403 - Fobidden), trying to delete current user (200 OK), trying to sign in (400 - Bad Request - User not found).

Green light from me.
